File: youqugou-interface/src/main/java/cn/itcast/core/service/UserService.java
package cn.itcast.core.service;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.user.User;

import java.util.List;
import java.util.Map;

public interface UserService {
    void sendCode(String phone);

    void add(String smscode, User user);

    PageResult searchUser(Integer currentPage, Integer itemsPerPage);

    void freezeUser(Long[] ids);

    List<Map> userCount();

    List<Integer> userActive();

    void addselfCenter(User user);

    User loginInfo(String username);
}

File: youqugou-web-manager/src/main/webapp/js/service/userService.js
app.service("userService", function ($http) {

    this.searchUser = function (currentPage, itemsPerPage) {
        return $http.get("../userManager/searchUser.do?currentPage="+currentPage+"&itemsPerPage="+itemsPerPage);
    }

    this.freeze = function (ids) {
        return $http.post("../userManager/freezeUser.do",ids);
    }

    this.userCount = function () {
        return $http.get("../userManager/userCount.do");
    }
});